src/misc/lv_array.h

```diff
@@ -38,46 +38,159 @@ typedef struct {
+
+/**
+ * Return if the array is empty
+ * @param array pointer to an `lv_array_t` variable
+ * @return true: array is empty; false: array is not empty
+ */
+bool lv_array_is_empty(const lv_array_t * array);
+
+/**
+ * Return if the array is full
+ * @param array pointer to an `lv_array_t` variable
+ * @return true: array is full; false: array is not full
+ */
+bool lv_array_is_full(const lv_array_t * array);
+
+/**
+ * Copy an array to another.
+ * @note this will create a new array with the same capacity and size as the source array.
+ * @param target pointer to an `lv_array_t` variable to copy to
+ * @param source pointer to an `lv_array_t` variable to copy from
+ */
+void lv_array_copy(lv_array_t * target, const lv_array_t * source);
 
+/**
+ * Remove all elements in array.
+ * @param array pointer to an `lv_array_t` variable
+ */
 void lv_array_clear(lv_array_t * array);
 
-void lv_array_resize(lv_array_t * array, uint32_t new_capacity);
+/**
+ * Remove the element at the specified position in the array.
+ * @param array pointer to an `lv_array_t` variable
+ * @param index the index of the element to remove
+ * @return true: success; false: error
+ */
+bool lv_array_pop(lv_array_t * array, uint32_t index);
 
-void lv_array_destroy(lv_array_t * array);
+/**
+ * Remove from the array either a single element or a range of elements ([start, end)).
+ * @note This effectively reduces the container size by the number of elements removed.
+ * @note When start equals to end, the function has no effect.
+ * @param array pointer to an `lv_array_t` variable
+ * @param start the index of the first element to be removed
+ * @param end the index of the first element that is not to be removed
+ * @return true: success; false: error
+ */
+bool lv_array_erase(lv_array_t * array, uint32_t start, uint32_t end);
 
-bool lv_array_append(lv_array_t * array, const uint8_t * element);
+/**
+ * Push back elements. Adds new elements to the end of the array.
+ * @param array pointer to an `lv_array_t` variable
+ * @param other pointer to the array to concatenate
+ * @return true: success; false: error
+ */
+bool lv_array_concat(lv_array_t * array, const lv_array_t * other);
 
-uint8_t * lv_array_get(const lv_array_t * array, uint32_t index);
+/**
+ * Concatenate two arrays. Adds new elements to the end of the array.
+ * @note The destination array must be large enough to hold the result.
+ * @param array pointer to an `lv_array_t` variable
+ * @param other pointer to the array to concatenate
+ * @return true: success; false: error
+ */
+bool lv_array_concat(lv_array_t * array, const lv_array_t * other);
 
-uint32_t lv_array_length(const lv_array_t * array);
+/**
+ * Push back element. Adds a new element to the end of the array.
+ * @param array pointer to an `lv_array_t` variable
+ * @param element pointer to the element to add
+ * @return true: success; false: error
+ */
+bool lv_array_push_back(lv_array_t * array, const void * element);
```

Comment:
To follow the name convention from other programming languages, I think the best name would be:
- `lv_array_append()`
- `lv_array_add()`

I prefer the first one, because it really means to add an element to the end of the array

Reply:
I'm following C++ STL, thought it's more popular to use. The original name is actually `_append`.
What do you think? Should we follow certain standard?

Reply:
Hmmm, I see.
Then keep STL naming.

Reply:
"append" sounds better to me too, but let's make it consistent with STL.